File: pkg/tunproxy/route_other.go
//go:build !linux && !darwin

package tunproxy

import (
	"fmt"

	"github.com/puppy/pkg/common"
)

type unsupportedHostNetworkManager struct{}

func newHostNetworkManager(device, ipv4Addr, ipv6Addr string, autoRoute bool) hostNetworkManager {
	return unsupportedHostNetworkManager{}
}

func (unsupportedHostNetworkManager) Apply() (common.Dialer, error) {
	return nil, fmt.Errorf("tunproxy: host network configuration not supported on this platform")
}

func (unsupportedHostNetworkManager) Restore() error { return nil }

func (unsupportedHostNetworkManager) EgressInterfaces() (string, string) { return "", "" }

File: rust/crates/tun/src/route_other.rs
//! Unsupported-platform stub for the host network manager.
//!
//! Mirrors Go `pkg/tunproxy/route_other.go`.

use std::io;

use async_trait::async_trait;

use crate::route::{DnsInterceptHandler, HostNetworkManager};
use puppy_core::backend::Dialer;
use std::sync::Arc;

/// Host network manager for platforms without native route support.
///
/// Mirrors Go `unsupportedHostNetworkManager` (pkg/tunproxy/route_other.go:11).
pub struct UnsupportedHostNetworkManager;

#[async_trait]
impl HostNetworkManager for UnsupportedHostNetworkManager {
	async fn apply(&mut self) -> io::Result<Arc<dyn Dialer>> {
		Err(io::Error::other(
			"tunproxy: host network configuration not supported on this platform",
		))
	}

	async fn enable_dns_interception(
		&mut self,
		_handler: Arc<dyn DnsInterceptHandler>,
	) -> io::Result<()> {
		Ok(())
	}

	async fn restore(&mut self) -> io::Result<()> {
		Ok(())
	}

	fn egress_interfaces(&self) -> (String, String) {
		(String::new(), String::new())
	}
}
